[ Monterey+ wk2 ] imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-GC.https.html is a flaky text failure

youennf opened the pull request:
#### e33391b688bac2c4e3374778e2314efd99735de2
<pre>
[ Monterey+ wk2 ] imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-GC.https.html is a flaky text failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=268098">https://bugs.webkit.org/show_bug.cgi?id=268098</a>
<a href="https://rdar.apple.com/121613695">rdar://121613695</a>

Reviewed by Eric Carlson.

On Intel x86, the VCP code path is not supporting resolution below 192x108.
We resort on SW code path using usage = 1, until the OS supports it.

* LayoutTests/platform/mac-wk2/TestExpectations:
* Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/RTCVideoEncoderH264.mm:
(-[RTCVideoEncoderH264 resetCompressionSessionWithPixelFormat:]):

Canonical link: <a href="https://commits.webkit.org/275243@main">https://commits.webkit.org/275243@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5d2c1bb148e2c0590fd7fed474f4023d11d51160

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41251 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20265 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43629 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/43815 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37343 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23345 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17595 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34114 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/41825 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17201 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35530 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14769 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/14931 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36532 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45140 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37448 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/36851 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40587 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16066 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13174 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/38972 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/17685 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9263 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/17737 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17329 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on LayoutTests/platform/mac-wk2/TestExpectations:
```diff
@@ -1968,11 +1968,10 @@ webkit.org/b/268018 [ Sonoma+ Release ] fast/images/text-recognition/mac/select-
 webkit.org/b/268200 [ Sonoma+ Release ] inspector/console/console-screenshot.html [ Pass Failure ]
 
-webkit.org/b/268332 [ Monterey+ x86_64 ] imported/w3c/web-platform-tests/webrtc/simulcast/getStats.https.html [ Failure ]
+webkit.org/b/268047 [ Sonoma+ x86_64 ] imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-GC.https.html [ Pass Failure ]
```
why pass failure ?

this isn't really testing the change, we allow to fail

youennf replied:
This fixes for Monterey and Ventura, I'll make it unconditional as discussed.

jyavenard commented on Source/ThirdParty/libwebrtc/Source/webrtc/sdk/WebKit/VideoProcessingSoftLink.h:
```diff
@@ -40,4 +40,10 @@
 #endif
 
+#if (defined(TARGET_OS_MAC) && TARGET_OS_MAC)
+#define ENABLE_LOW_LATENCY_INTEL_ENCODER_FOR_LOW_RESOLUTION (__MAC_OS_X_VERSION_MIN_REQUIRED >= 140500)
```
shouldn't this be set in regards to ARCH family here instead?

youennf replied:
There is the name INTEL in the macro

eric-carlson commented on Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/RTCVideoEncoderH264.mm:
```diff
@@ -804,9 +804,18 @@ - (int)resetCompressionSessionWithPixelFormat:(OSType)framePixelFormat {
+  if (_isH264LowLatencyEncoderEnabled && _useVCP) {
+#if defined(WEBRTC_ARCH_X86_FAMILY) && !ENABLE_LOW_LATENCY_INTEL_ENCODER_FOR_LOW_RESOLUTION
+    if (_width < 192 || _height < 108) {
+      int usageValue = 1;
+      auto usage = CFNumberCreate(nullptr, kCFNumberIntType, &usageValue);
+      CFDictionarySetValue(encoderSpecs, kVTCompressionPropertyKey_Usage, usage);
+      CFRelease(usage);
```
Could you use `CFDictionarySetValue(encoderSpecs, kVTCompressionPropertyKey_Usage, @( 1 ))`  instead of creating and releasing the CFNumber?

youennf replied:
This appears to not be supported by CFDictionarySetValue unfortunately

youennf committed:
[ Monterey+ wk2 ] imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-GC.https.html is a flaky text failure
https://bugs.webkit.org/show_bug.cgi?id=268098
rdar://121613695

Reviewed by Eric Carlson.

On Intel x86, the VCP code path is not supporting resolution below 192x108.
We resort on SW code path using usage = 1, until the OS supports it.

* LayoutTests/platform/mac-wk2/TestExpectations:
* Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/RTCVideoEncoderH264.mm:
(-[RTCVideoEncoderH264 resetCompressionSessionWithPixelFormat:]):

Canonical link: https://commits.webkit.org/275243@main
